session.go

```diff
@@ -488,14 +533,22 @@ func (s *session) Close() error {
 		if s.beforeClose != nil {
 			s.beforeClose(s)
 		}
-		err = s.Conn.Close()
```

Comment:
This now happens when the recvLoop terminates.

Reply:
`err` should be set to `nil` somewhere, or `errorAlreadyClosed` will be returned even it's called for the first time.

Reply:
Good catch! Fixed.